datafusion/physical-expr/src/aggregate/first_last.rs

```diff
@@ -112,25 +112,35 @@ impl PartialEq<dyn Any> for FirstValue {
 #[derive(Debug)]
 struct FirstValueAccumulator {
     first: ScalarValue,
+    // At the beginning, `is_set` is `false`, this means `first` is not seen yet.
+    // Once we see (`is_set=true`) first value, we do not update `first`.
+    is_set: bool,
```

Comment:
Another potential way that might let the compiler check that `is_set` was updated correctly would be something like:

```
struct FirstValueAccumulator {
  /// None until the data has been seen
  first: Option<ScalarValue>
}

```

Reply:
I have considered this option, however, during `evaluate` call if `first` is `None`. We need to return `NULL` for appropriate type, such as `ScalarValue::Int64(None)` if type is `Int64`. However, datatype is only available during initialization. If we want to use `Option` for `first`, we may need to store data type also. Hence I opted for this approach